src/Service/Resolvers/QueryExecutor.cs

```diff
@@ -154,16 +155,20 @@ public class QueryExecutor<TConnection> : IQueryExecutor
             // Add query to send user data from DAB to the underlying database to enable additional security the user might have configured
             // at the database level.
             string sessionParamsQuery = GetSessionParamsQuery(httpContext, parameters);
-            //"EXEC sp_set_session_context 'roles', 'Anonymous', @read_only =1 ;";
 
             cmd.CommandText = sessionParamsQuery + sqltext;
             if (parameters is not null)
             {
-                foreach (KeyValuePair<string, object?> parameterEntry in parameters)
+                foreach (KeyValuePair<string, Tuple<object?, DbType?>> parameterEntry in parameters)
                 {
                     DbParameter parameter = cmd.CreateParameter();
                     parameter.ParameterName = parameterEntry.Key;
-                    parameter.Value = parameterEntry.Value ?? DBNull.Value;
+                    parameter.Value = parameterEntry.Value.Item1 ?? DBNull.Value;
+                    if (parameterEntry.Value.Item2 is not null && ConfigProvider.GetRuntimeConfiguration().DatabaseType is DatabaseType.mssql)
```

Comment:
Even though other relational dbs dont require the type specification at the moment, they may still benefit from it. Do the tests fail if we provide the parameter type? 

Adding dbchecks in a common function seems counter intuitive, 
if we have to deviate the implementation, I think its better to override the implementation in the MsSql specific executor

Reply:
Yes the tests fail because of the intricacies of each databases.

Reply:
Just the condition to whether add the DbType or not is different among the DBs right now. Extracted it out into a different overridden function.